### PR TITLE
Migrates NServiceBus 5 tests to use ConsoleMF app

### LINKED ISSUE #820: Migrate existing NServiceBus 5 tests to ConsoleMF
### Feature Description

There were issues in the past with migrating the tests that need to be rediscovered and resolved as part of this work.
Some metrics will likely need to be changed as well.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Migrates existing NServiceBus apps, class, and libraries to use the ConsoleMF framework app.
* Removes projects: NServiceBusReceiver, NServiceBusReceiverHost, NServiceBusBasicMvcApplication
* Removes fixtures: NServiceBusBasicMvcApplicationFixture and NServiceBusReceiverFixture

New NServiceBus5 libraries in MultiFunctionApplicationHelpers handle everything for NSB now.
* NSB 5 is only available in net462.  NSB5 is not supported but Particular anymore and so this limitation allows new versions to make use of the new framework versions for testing. Class have been placed behind preprocessor directives.
* Added a setup method to ensure everything is in place before running tests
* Only the default Send command is manually instrumented as a transaction since the receive side will create one for us.

Add a RootCommands Library that current only has a DelaySeconds command used to allow messages to arrive without using AgentLog specific delays (might want revisit later)

Testing the Send/Receive using consoleMF outside of the test suite
Start the ConsoleMF app for framework

Send:
NServiceBusSetup Setup
NServiceBusService Start
NServiceBusService  Send

Result:  
"Message with ID={message.Id} sent via NServiceBus"

----

Send/Receive 
NServiceBusSetup Setup
NServiceBusService Start
NServiceBusReceiverHost Start
NServiceBusService  SendValid
NServiceBusService  SendInvalid

Result:
Message with ID={message.Id} sent via NServiceBus
Message with ID={message.Id} sent via NServiceBus
Received Valid message with contents={message.FooBar}
Received Invalid message with contents={message.FooBar}
Message was invalid, throwing an exception!
An exception was thrown inside the NServiceBus Receive Handler!!!!

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

resolves #820

### CODE REVIEW

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/RootCommands.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using System.Threading.Tasks;
+using NewRelic.Agent.IntegrationTests.Shared.ReflectionHelpers;
+
+namespace MultiFunctionApplicationHelpers
+{
+    [Library]
+    public static class RootCommands
+    {
+        [LibraryMethod]
+        public static void DelaySeconds(int seconds)
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/RootCommands.cs:15]: Nice! Surprised we didn't have this

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            var expectedTransactionTraceSegments = new List<string>
+            {
+                @"MessageBroker/NServiceBus/Queue/Produce/Named/MultiFunctionApplicationHelpers.NetStandardLibraries.NServiceBus5.SampleNServiceBusMessage"
+            };
+
+            var metrics = _fixture.AgentLog.GetMetrics().ToList();
+            var transactionSample = _fixture.AgentLog.TryGetTransactionSample("OtherTransaction/Custom/MultiFunctionApplicationHelpers.NetStandardLibraries.NServiceBus5.NServiceBusService/Send");
+            var transactionEvent = _fixture.AgentLog.TryGetTransactionEvent("OtherTransaction/Custom/MultiFunctionApplicationHelpers.NetStandardLibraries.NServiceBus5.NServiceBusService/Send");
+
+            NrAssert.Multiple(
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs:60]: I'm not a fan of NrAssert.Multiple(...), but this is consistent with the rest of our codebase :)

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand("NServiceBusSetup Setup");
+            _fixture.AddCommand("NServiceBusReceiverHost Start");
+            _fixture.AddCommand("NServiceBusService Start");
+            _fixture.AddCommand("NServiceBusService SendValid");
+            _fixture.AddCommand("NServiceBusService SendInvalid");
+            _fixture.AddCommand("RootCommands DelaySeconds 5");
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs:32]: Hopefully this is reliable. I could see this being a source of issues on slow machines/the cloud

**Review by JcolemanNR — APPROVED** [review]
Looks good!

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/RootCommands.cs (hunk 1 above)
Comment (jaffinito) [reply tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/RootCommands.cs:15]: Yeah, I could have sworn we did at one point.

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs (hunk 2 above)
Comment (jaffinito) [reply tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs:60]: NrAssert.Multiple does have the benefit of allowing the test to proceed and examine all the contained asserts before failing.  I am in favor of leaving them in, at least for now.   You cool with that?

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs (hunk 3 above)
Comment (jaffinito) [reply tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs:32]: I was worried about that too, but I was having trouble with the AgentLog taking forever and needing more than one entry to for the various data types.  I would prefer to extend this style of wait-timer out a bit longer over going back to the AgentLog watcher. 

Thoughts?

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs (hunk 2 above)
Comment (JcolemanNR) [reply tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5Tests.cs:60]: Sounds good!

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs (hunk 3 above)
Comment (JcolemanNR) [reply tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs:32]: Let's see how reliable it is, and extend if needed?

**Pull request merged**